File: Java/dce/src/main/java/com/rgbeans/dce300/Vote.java
package com.rgbeans.dce300;

public record Vote(int voterID, int candidateID) {
    public static boolean tryParse(String text, Vote[] result) {
        result[0] = new Vote(-1, -1);

        String[] parts = text.split(",");

        if (parts.length != 2) {
            return false;
        }

        try {
            int voterID = Integer.parseInt(parts[0]);
            int candidateID = Integer.parseInt(parts[1]);
            result[0] = new Vote(voterID, candidateID);
        } catch (NumberFormatException e) {
            return false;
        }

        return true;
    }
}

File: CSharp/DceLib/Dce300/Vote.cs
namespace DceLib.Dce300;

public record Vote (int voterID, int candidateID) {
    public static bool TryParse (string text, out Vote result)
    {
        result = new Vote(-1,-1);

        string[] parts = text.Split(',');

        if (parts.Length != 2)
        {
            return false;
        }

        if (!int.TryParse(parts[0], out int voterID))
        {
            return false;
        }

        if (!int.TryParse(parts[1], out int candidateID))
        {
            return false;
        }

        result = new Vote(voterID, candidateID);

        return true;
    }
}

public record Result (int topCandidate, int secondCandidate, int thirdCandidate, bool fraud);